### PR TITLE
MGMT-7716: Add a template file for the image service

### PR DESCRIPTION
**Pull request by carbonin**
## Description

This will allow us to deploy the image service to the SaaS and will also allow
us to deploy new versions of the image service in the same way we do for the
assisted service.

Specifically this needs to be a separate template (rather than a new
deployment in the existing one) so that the app-sre automation can
handle automatically deploying it on changes to upstream repo.
If we used the same template we would need to manually update the tag
for the image service image we wanted to deploy in saas.yaml each time
it changed.

https://issues.redhat.com/browse/MGMT-7716

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

* Deployed assisted-service to minikube with `make deploy-all`
* Deployed the image service to minikube with:
`oc process --local -f openshift/template-image-service.yaml -p OS_IMAGES='[{"openshift_version":"4.8","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.8/4.8.2/rhcos-4.8.2-x86_64-live.x86_64.iso","rootfs_url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.8/4.8.2/rhcos-live-rootfs.x86_64.img","version":"48.84.202107202156-0"}]' -p IMAGE_TAG=latest -p REPLICAS_COUNT=1 | kubectl -n assisted-installer apply -f -`
* Exposed the image service by changing the image service `Service` object to a `LoadBalancer` type (like the others when `make deploy-all` is used)
* Created an infraenv
* Downloaded the image for that infraenv from the image service
* Validated the ignition was correct

## Assignees

/cc @filanov 
/cc @osherdp 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

### CODE REVIEW

File: openshift/template-image-service.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+parameters:
+- name: IMAGE_SERVICE_IMAGE
+  value: quay.io/app-sre/assisted-image-service
+- name: IMAGE_TAG
+  value: ''
+  required: true
+- name: OS_IMAGES
+  value: '[{"openshift_version":"4.6","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.8/rhcos-4.6.8-x86_64-live.x86_64.iso","rootfs_url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.8/rhcos-live-rootfs.x86_64.img","version":"46.82.202012051820-0"},{"openshift_version":"4.7","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.7/4.7.13/rhcos-4.7.13-x86_64-live.x86_64.iso","rootfs_url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.7/4.7.13/rhcos-live-rootfs.x86_64.img","version":"47.83.202105220305-0"},{"openshift_version":"4.8","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.8/4.8.2/rhcos-4.8.2-x86_64-live.x86_64.iso","rootfs_url":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.8/4.8.2/rhcos-live-rootfs.x86_64.img","version":"48.84.202107202156-0"}]'
```
[the remark this answers is not in the document]
Comment (carbonin) [reply openshift/template-image-service.yaml:9]: Maybe? Like said before, I just copied it.
Also I think we can probably just remove this.

[the remark this answers is not in the document]
Comment (carbonin) [reply openshift/template-image-service.yaml]: > Also I think we can probably just remove this.

This is not true. This is the value we use in integration.

File: openshift/template-image-service.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: assisted-image-service
+  spec:
+    selector:
+      matchLabels:
+        service: image-service
+    replicas: ${{REPLICAS_COUNT}}
```
[the remark this answers is not in the document]
Comment (carbonin) [reply openshift/template-image-service.yaml:26]: Also described in the openshift docs for templates as used for the unquoted value: https://docs.openshift.com/container-platform/4.8/openshift_images/using-templates.html#templates-writing-parameters_using-templates

File: openshift/template-image-service.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (osherdp) [reply openshift/template-image-service.yaml]: ok, so leave it as-is

**Review by osherdp — APPROVED** [review]

**Review by filanov — APPROVED** [review]

File: openshift/template-image-service.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          - name: assisted-image-service
+            image: ${IMAGE_SERVICE_IMAGE}:${IMAGE_TAG}
+            resources:
+              requests:
+                cpu: 100m
+                memory: 400Mi
+            ports:
+              - protocol: TCP
+                containerPort: 8080
+            readinessProbe:
```
[the remark this answers is not in the document]
Comment (carbonin) [reply openshift/template-image-service.yaml:43]: Yup, opened this before it was merged :sweat_smile:

**Pull request merged**